File: backend/app/schemas/reporting.py
from datetime import date, datetime
from uuid import UUID

from pydantic import BaseModel, Field


class ChatHistoryItem(BaseModel):
    session_id: UUID
    user_label: str
    started_at: datetime
    ended_at: datetime | None
    response_count: int
    completed_count: int
    failed_count: int
    faq_count: int
    generated_ai_count: int
    no_answer_count: int
    good_count: int
    bad_count: int


class ChatHistoryResponse(BaseModel):
    from_date: date
    to_date: date
    page: int = Field(ge=1)
    page_size: int = Field(ge=1, le=100)
    total_count: int
    items: list[ChatHistoryItem]

File: frontend/types/reporting.ts
export type ChatHistoryItem = {
  session_id: string;
  user_label: string;
  user_id: string | null;
  user_name: string | null;
  user_role: string | null;
  user_site: string | null;
  started_at: string;
  ended_at: string | null;
  response_count: number;
  completed_count: number;
  failed_count: number;
  faq_count: number;
  generated_ai_count: number;
  no_answer_count: number;
  good_count: number;
  bad_count: number;
};

export type ChatHistoryResponse = {
  from_date: string;
  to_date: string;
  page: number;
  page_size: number;
  total_count: number;
  items: ChatHistoryItem[];
};
